pyomo/core/base/units_container.py

```diff
@@ -1408,28 +1419,17 @@ def convert(self, src, to_units=None):
         """
         src_pyomo_unit, src_pint_unit = self._get_units_tuple(src)
         to_pyomo_unit, to_pint_unit = self._get_units_tuple(to_units)
-
-        # check if any units have offset
-        # CDL: This is no longer necessary since we don't allow
-        # offset units, but let's keep the code in case we change
-        # our mind about offset units
-        #  src_unit_container = pint.util.to_units_container(src_unit, self._pint_ureg)
-        # dest_unit_container = pint.util.to_units_container(dest_unit, self._pint_ureg)
-        # src_offset_units = [(u, e) for u, e in src_unit_container.items()
-        #                     if not self._pint_ureg._units[u].is_multiplicative]
-        # 
-        #  dest_offset_units = [(u, e) for u, e in dest_unit_container.items()
-        #                 if not self._pint_ureg._units[u].is_multiplicative]
-
-        # if len(src_offset_units) + len(dest_offset_units) != 0:
-        #     raise UnitsError('Offset unit detected in call to convert. Offset units are not supported at this time.')
+        
+        if src_pyomo_unit is None and to_pyomo_unit is None:
+            return src
 
         # no offsets, we only need a factor to convert between the two
         fac_b_src, base_units_src = self._pint_registry.get_base_units(src_pint_unit, check_nonmult=True)
         fac_b_dest, base_units_dest = self._pint_registry.get_base_units(to_pint_unit, check_nonmult=True)
 
         if base_units_src != base_units_dest:
-            raise UnitsError('Cannot convert {0:s} to {1:s}. Units are not compatible.'.format(str(src_pyomo_unit), str(to_pyomo_unit)))
+            raise InconsistentUnitsError(src_pint_unit, to_pint_unit,
+                                             'Error in convert: units not compatible.')
```

Comment:
Would it make sense to add the units to the exception message?

Also, silly thing, but it looks like there is a weird indentation for the second line.

Reply:
The units are included in the exception message (they are passed in as the first two arguments, and added to the message in InconsistentUnitsError). I can look at the indentation. Have we ever standardized on what the indentation for continuation stuff should be?

Reply:
Perfect.

As for standards, nope.  My editor (emacs) aligns things within the parens (so the "`'`" character would be directly under the '`s`' from `src_pint_unit`).